catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/CatblocksScriptFragmentTest.java

```diff
@@ -0,0 +1,204 @@
+
+	@Rule
+	public FragmentActivityTestRule<SpriteActivity> baseActivityTestRule = new
+			FragmentActivityTestRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION,
+			SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@Before
+	public void setUp() throws Exception {
+		SettingsFragment.setUseCatBlocks(ApplicationProvider.getApplicationContext(), false);
+		createProject();
+		baseActivityTestRule.launchActivity();
+	}
+
+	@After
+	public void tearDown() {
+		baseActivityTestRule.getActivity().finish();
+	}
+
+	@Test
+	public void testContextMenuItems() {
+
+		openContextualActionModeOverflowMenu();
+
+		// default is 1D view -> reorder not visible
+		onView(withText(R.string.catblocks_reorder)).check(doesNotExist());
+
+		onView(withText(R.string.catblocks)).perform(click());
+
+		// in Catblocks reorder should be visible
+		openContextualActionModeOverflowMenu();
+
+		onView(withText(R.string.catblocks_reorder)).check(matches(isDisplayed()));
+
+		onView(withText(R.string.undo)).check(doesNotExist());
+		onView(withText(R.string.backpack)).check(doesNotExist());
+		onView(withText(R.string.copy)).check(doesNotExist());
+		onView(withText(R.string.delete)).check(doesNotExist());
+		onView(withText(R.string.rename)).check(doesNotExist());
+		onView(withText(R.string.show_details)).check(doesNotExist());
+		onView(withText(R.string.comment_in_out)).check(doesNotExist());
+
+		onView(withText(R.string.catblocks)).perform(click());
+
+		// change back to 1D and check for existing menu items
+		openContextualActionModeOverflowMenu();
+
+		onView(withText(R.string.catblocks_reorder)).check(doesNotExist());
+
+		onView(withText(R.string.backpack)).check(matches(isDisplayed()));
+		onView(withText(R.string.copy)).check(matches(isDisplayed()));
+		onView(withText(R.string.comment_in_out)).check(matches(isDisplayed()));
+		onView(withText(R.string.catblocks)).check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void testCorrect2DObjectDispayed() {
+		openContextualActionModeOverflowMenu();
+		onView(withText(R.string.catblocks)).perform(click());
+
+		int loadCounter = 0;
+		// since rendering may take some time: try multiple times
+		while (true) {
+			loadCounter++;
+			try {
+				onWebView().check(webContent(hasElementWithXpath("//*[@data-object=\"testSprite\" "
+						+ "and contains(@class,'catblocks-script-container') "
+						+ "and contains(@class,'collapse') "
+						+ "and contains(@class,'show')"
+						+ "]")));
+				break;
+			} catch (Throwable e) {
+				// max. 2.5 seconds - then timeout
+				if (loadCounter >= 25) {
+					throw e;
+				}
+				try {
+					Thread.sleep(100);
+				} catch (InterruptedException ex) {
+					// ignore
+				}
+			}
+		}
+	}
+
+	@Test
+	public void testReorderScript() {
+		openContextualActionModeOverflowMenu();
+		onView(withText(R.string.catblocks)).perform(click());
+
+		int loadCounter = 0;
+		// since rendering may take some time: try multiple times if finished
+		while (true) {
```

Comment:
While loop is duplicate of test above. Do we need to check this again? Maybe delete in this test, else delete test above.